src/server/req_manager.c

```diff
@@ -3109,6 +3121,54 @@ check_sister_vnodes_for_delete(mom_svrinfo_t *psvrmom)
 	return (ct);
 }
 
+/**
+ *  @brief delete a scheduler object
+ *
+ *  @param[in] preq - Pointer to a batch request structure
+ *
+ */
+static void
+mgr_sched_delete(struct batch_request *preq)
+{
+	pbs_sched *psched;
+	pbs_sched *tmpsched;
+
+	if ((*preq->rq_ind.rq_manager.rq_objname == '\0')
+			|| (*preq->rq_ind.rq_manager.rq_objname == '@')) {
+		/* Delete all except default */
+		psched = (pbs_sched *) GET_NEXT(svr_allscheds);
+		while (psched != (pbs_sched *) 0) {
+			tmpsched = psched;
+			psched = (pbs_sched *) GET_NEXT(psched->sc_link);
+			if (tmpsched != dflt_scheduler) {
+				if (sched_delete(tmpsched) == PBSE_OBJBUSY) {
+					snprintf(log_buffer, sizeof(log_buffer),
+							"Scheduler %s is busy", tmpsched->sc_name);
+					log_err(PBSE_OBJBUSY, __func__, log_buffer);
+				}
+			}
+		}
+	} else {
+		psched = find_scheduler(preq->rq_ind.rq_manager.rq_objname);
+		if (!psched) {
+			req_reject(PBSE_UNKSCHED, 0, preq);
+			return;
+		} else if (psched == dflt_scheduler) {
+			req_reject(PBSE_SCHED_NO_DEL, 0, preq);
+			return;
+		}
+
+		if (sched_delete(psched) == PBSE_OBJBUSY) {
+			snprintf(log_buffer, sizeof(log_buffer),
+					"Scheduler %s is busy", psched->sc_name);
+			log_err(PBSE_OBJBUSY, __func__, log_buffer);
+			(void) reply_text(preq, PBSE_OBJBUSY, preq->rq_ind.rq_manager.rq_objname);
```

Comment:
you still need to return after reply_text because reply_text and reply_ack both these functions will try to reply back to the client and then call free on preq